### PR TITLE
Password Reset

### PR DESCRIPTION
**Pull request by ryancj**

### CODE REVIEW

**Comment by zibs** [comment]
We'll have to add tests later, but +1 nonetheless

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,8 @@
 # Ignore bundler config.
 /.bundle
 
+/config/initializers/app_keys.rb
```
Comment (tkbeili) [inline .gitignore:10]: This seems to be repeated in another PR. Make sure to rebase once of the PR is merged.

File: app/controllers/password_resets_controller.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    else
+      render file: 'public/404.html', status: :not_found
+    end
+  end
+
+  def update
+    @user = User.find_by_password_reset_token params[:id]
+    if @user && @user.update_attributes(user_params)
+      @user.update_attribute(:password_reset_token, nil)
+      session[:user_id] = @user.id
```
Comment (tkbeili) [inline app/controllers/password_resets_controller.rb]: I suggest creating a `login` method and put it in the application controller

File: app/controllers/password_resets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      render action: 'new'
+    end
+  end
+
+  def edit
+    @user = User.friendly.find_by_password_reset_token params[:id]
+    if @user
+    else
+      render file: 'public/404.html', status: :not_found
+    end
```
Comment (tkbeili) [inline app/controllers/password_resets_controller.rb]: This can be written as:

``` ruby
render file: 'public/404.html', status: :not_found unless @user
```

**Comment by tkbeili** [comment]
Please fix the notes I've put and add specs.

**Comment by ryancj** [comment]
Adjusted the code to Tam's suggestions, also added tests for user model and password reset controller.

**Comment by zibs** [comment]
+1

**Comment by tkbeili** [comment]
:+1:
